contracts/payment/MCPayment.sol

```diff
@@ -135,12 +186,47 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
                 keccak256(paymentData.metadata)
             )
         );
+        if (!isSignatureValid(structHash, signature, paymentData.recipient)) {
+            revert InvalidSignature("MCPayment: invalid signature for Iden3PaymentRailsRequestV1");
+        }
+    }
+
+    function verifyERC20Signature(
+        Iden3PaymentRailsERC20RequestV1 memory paymentData,
+        bytes memory signature
+    ) public view {
+        bytes32 structHash = keccak256(
+            abi.encode(
+                ERC_20_PAYMENT_DATA_TYPE_HASH,
+                paymentData.tokenAddress,
+                paymentData.recipient,
+                paymentData.amount,
+                paymentData.expirationDate,
+                paymentData.nonce,
+                keccak256(paymentData.metadata)
+            )
+        );
+
+        if (!isSignatureValid(structHash, signature, paymentData.recipient)) {
+            revert InvalidSignature(
+                "MCPayment: invalid signature for Iden3PaymentRailsERC20RequestV1"
+            );
+        }
+    }
+
+    function isSignatureValid(
```

Comment:
better to use "_" in the beginning of all internal and private functions. Easier to real function lists.